mobile_app/openSleep/Recordings.swift

```diff
@@ -180,38 +254,196 @@ class RecordingsManager : NSObject, AVAudioRecorderDelegate, AVAudioPlayerDelega
     }
   }
   
-  func startRecordingDream(dreamTitle: String) {
+  // To eyal: Right now, startRecording and startRecordingulti both exist, I could combine them, to allow for multiple wakeup prompts, or leave this as is
+  /*
+   Called to record multiple Remember to think of messages. Appends urls to the audioMultiUrls[0] array
+ */
+  func startRecordingMulti(mode: Int) {
     let audioSession = AVAudioSession.sharedInstance()
+    
     do {
-      if let url = self.audioDirectoryURLwithTimestamp() {
-        addRecording(categoryName: dreamTitle, path: url.absoluteString, length: 60)
+      if let url = self.audioDirectoryURLMulti(mode) {
         audioRecorder = try AVAudioRecorder(url: url as URL,
                                             settings: audioRecorderSettings)
         audioRecorder.delegate = self
         audioRecorder.prepareToRecord()
         
-        print("url = \(url)")
+        audioMultiURLs[mode]!.append(url)
+        print("Multi url = \(url)")
+      }
+    }  catch {
+      audioRecorder.stop()
+    }
+    do {
+      try audioSession.setActive(true)
+      audioRecorder.record()
+    } catch {
+    }
+  }
+  
+  // To eyal: Wondering if this is the best way to handle silence detection
+  // In particular, I feel like the silenceCallback is a little clunky
```

Comment:
I think it's ok, up to you. If we expect his silence logic to exist in other places in the code, then maybe externalize and delegate. But for now I think it's fine